qtwebengine/src/core/content_browser_client_qt.cpp

```diff
@@ -217,24 +222,44 @@ class MessagePumpForUIQt : public QObject,
 
 private:
     bool handleScheduledWork() {
```

Comment:
Would it be clearer if we did something like:

```
bool handleScheduledWork() {
   for(unsigned int num_completed = 0; num_completed < m_eventCount; num_completed++)
   {
      if (!_handleScheduledWork())
      {
         return false;
      }
   }
   return true;
}
```

Basically move their loop method to a slightly different name and then try running it x number of times when we get called.